README.md

```diff
@@ -5,757 +5,875 @@
-// => { t1: Todo { title: 'Buy milk', isComplete: false }, t2: Todo { title: 'Review PRs', isComplete: false } }
+class LightMachine {
+  color = String;
+
+  initialize({ color: 'green' } = {}) {
+    return this.color.set(color);
+  }
 
-today.t2.isComplete.toggle().state
-// => { t1: Todo { title: 'Buy milk', isComplete: false }, t2: Todo { title: 'Review PRs', isComplete: true } }
+  timer() {
+    switch (this.color.state) {
+      case 'green': return this.color.set('yellow');
+      case 'yellow': return this.color.set('red');
+      case 'red':
+      default:
+        return this.color.set('green'); 
+    }
+  }
+}
 ```
 
-## Observable interoperability
+With Microstates, you explicitely describe what happens on transition and define the matching mechanism. 
 
-You can create an observable from any microstate using the `Observable.from` method.
-The resulting Observable will stream the next microstate for every transition.
-When you subscribe to the observable microstate, you'll immediately receive a microstate through the stream.
-This microstate will have transitions that you can call to cause the next microstate to come through the stream.
+### No transitionTo function
 
-```js
-import { create } from 'microstates';
-import { Observable } from 'rxjs';
+`transitionTo` is often used by state machine libraries to trigger state transition. Here is an example with xstate library,
 
-let ms = create(Number, 42);
-let observable = Observable.from(ms);
-
-let last;
-
-observable.subscribe(microstate => {
-  last = microstate;
-  console.log(microstate.state);
-});
-// => 42
+```js
+const nextState = lightMachine.transition('green', 'TIMER').value;
 
-last.increment().increment().increment();
-// => 43
-// => 44
-// => 45
+//> 'yellow'
 ```
 
-## Built-in types
+Microstates does not have such a method. Instead, the next state is derived by invoking a method on the microstate.
 
-Microstates package provides base building blocks for your composing your own types. `Number`, `String`, `Boolean`,
-`Object` and `Array` come with predefined transitions.
+```js
+import { create } from 'microstates';
 
-### `Boolean`
+let lightMachine = create(LightMachine);
 
-`Boolean` type presents a `true` or `false` value. `Boolean` has `toggle` and `set` transitions.
+const nextState = lightMachine.timer();
 
-#### set(value: any) => microstate
+nextState.color.state
+//> 'yellow'
+``` 
 
-Returns a new microstate with boolean value replaced. Value will be coerced with
-`Boolean(value).valueOf()`.
+## Immutable Object vs Immutable Data Structure
 
-```js
-Microstate.create(Boolean).set(true).state;
-// => true
-```
+When you create a state machine with xstate, you create an immutable object. When you invoke a transition on an `xstate` state machine, the value of the object is the ID of the next state. All of the concerns of immutable value change as a result of state change are left for you to handle manually.
```

Comment:
This is a big problem, and should get a higher billing. We should drive in the point that we're marrying value and state because a state machine is worthless if you can't actually store values. I think an example of this would be really key. Maybe actually model the passage of time with a `tick` method?

Reply:
Can you think of an example for this? ( I can write around it )

Reply:
I think actually having the time tracking in there. Does it make the example too complicated?